collections/soldiers.js

```diff
@@ -0,0 +1,47 @@
+const MongoClient = require('mongodb').MongoClient;
+const assert = require('assert');
+const configs = require('../config');
+const utils = require("../utility/utility");
+
+// Connection URL
+const url = configs.dbUrl;
+
+// Database Name
+const dbName = configs.dbName;
+
+// Use connect method to connect to the server
+function newSoldier(collection, data, errorHandler) {
+  if (data["id"] && !(utils.isMatchesRegex(/[0-9]{7}/, data["id"]))) {
+    errorHandler.notValidData("The soldier json that was sent didnt had a legal id");
+  } else {
```

Comment:
no need for this else, if the above thread happend the thread would have left this place